### PR TITLE
feat: implement `parseTomlType` cheats

### LINKED ISSUE #8810: feat(`cheatcodes`): add `vm.parseTomlType`
### Component

Forge

### Describe the feature you would like

In https://github.com/foundry-rs/foundry/pull/8345, `parseJsonType` was added. We should also add a corresponding `parseTomlType` cheat

I also wonder if we can support the string being the name of the struct directly, perhaps of the form `ContractName:StructName` or similar so it can be read from artifacts

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by leovct**
## Motivation

Closes #8810 

## Solution

Mostly a copy/paste of #8345.

Minor: removal of references of `json` in `testdata/default/cheats/Toml.t.sol` and use of `toml` instead.

### CODE REVIEW

File: testdata/default/cheats/Toml.t.sol [hunk 1]
```diff
@@ -225,48 +299,80 @@ contract ParseTomlTest is DSTest {
...
 
     function test_serializeNotSimpleToml() public {
-        string memory json3 = "json3";
+        string memory toml3 = "toml3";
         string memory path = "fixtures/Toml/write_complex_test.toml";
-        vm.serializeUint(json3, "a", uint256(123));
-        string memory semiFinal = vm.serializeString(json3, "b", "test");
-        string memory finalJson = vm.serializeString(json3, "c", semiFinal);
-        console.log(finalJson);
-        vm.writeToml(finalJson, path);
```
Comment (zerosnacks) [inline testdata/default/cheats/Toml.t.sol:259]: I originally used the `json` naming here to indicate that during operations users are operating on JSON data, it only becomes TOML once written

I would prefer this to be reverted but open to feedback

**Review by zerosnacks — COMMENTED** [review]
Thanks this is great! 

Minor feedback in regards to naming change in test

File: testdata/default/cheats/Toml.t.sol (hunk 1 above)
Comment (leovct) [reply testdata/default/cheats/Toml.t.sol:259]: That makes sense, I'll revert the changes

**Review by zerosnacks — APPROVED** [review]

File: testdata/default/cheats/Toml.t.sol (hunk 1 above)
Comment (leovct) [reply testdata/default/cheats/Toml.t.sol]: I've kept the `struct` renaming because I think it makes more sense, but I'm open to change :)

**Comment by DaniPopes** [comment]
PTAL @klkvr

**Review by klkvr — APPROVED** [review]

**Pull request merged**